drivers/clock_control/clock_stm32_ll_h7.c

```diff
@@ -501,163 +501,163 @@ static struct clock_control_driver_api stm32_clock_control_api = {
-		LL_RCC_PLL3R_Enable();
-		LL_RCC_PLL3_SetR(STM32_PLL3_R_DIVISOR);
-	}
+	if (IS_ENABLED(STM32_HSE_ENABLED)) {
+		/* Enable HSE oscillator */
+		if (IS_ENABLED(STM32_HSE_BYPASS)) {
+			LL_RCC_HSE_EnableBypass();
+		} else {
+			LL_RCC_HSE_DisableBypass();
+		}
 
-	LL_RCC_PLL3_Enable();
-	while (LL_RCC_PLL3_IsReady() != 1U) {
+		LL_RCC_HSE_Enable();
+		while (LL_RCC_HSE_IsReady() != 1) {
+		}
 	}
-#endif /* STM32_PLL3_ENABLED */
 
-	return 0;
-}
-
-__unused
-static void set_up_hse(void)
-{
-#if defined(STM32_SYSCLK_SRC_HSE) || defined(STM32_PLL_SRC_HSE)
-	if (IS_ENABLED(STM32_HSE_BYPASS)) {
-		LL_RCC_HSE_EnableBypass();
-	} else {
-		LL_RCC_HSE_DisableBypass();
+	if (IS_ENABLED(STM32_HSI_ENABLED)) {
+		/* Enable HSI oscillator */
+		LL_RCC_HSI_Enable();
+		while (LL_RCC_HSI_IsReady() != 1) {
+		}
+		/* HSI divider configuration */
+		LL_RCC_HSI_SetDivider(hsi_divider(STM32_HSI_DIVISOR));
 	}
 
-	/* Enable HSE oscillator */
-	LL_RCC_HSE_Enable();
-	while (LL_RCC_HSE_IsReady() != 1) {
+	if (IS_ENABLED(STM32_CSI_ENABLED)) {
+		/* Enable CSI oscillator */
+		LL_RCC_CSI_Enable();
+		while (LL_RCC_CSI_IsReady() != 1) {
+		}
 	}
-#endif /* STM32_*_SRC_HSE */
-}
 
-__unused
-static void set_up_hsi(void)
-{
-#if defined(STM32_SYSCLK_SRC_HSI) || defined(STM32_PLL_SRC_HSI)
-	/* Enable HSI oscillator */
-	LL_RCC_HSI_Enable();
-	while (LL_RCC_HSI_IsReady() != 1) {
+	if (IS_ENABLED(STM32_LSI_ENABLED)) {
+		/* Enable LSI oscillator */
+		LL_RCC_LSI_Enable();
+		while (LL_RCC_LSI_IsReady() != 1) {
+		}
 	}
 
-	/* HSI divider configuration */
-	LL_RCC_HSI_SetDivider(hsi_divider(STM32_HSI_DIVISOR));
-#endif /* STM32_*_SRC_HSI */
-}
+	if (IS_ENABLED(STM32_LSE_ENABLED)) {
+		/* Enable backup domain */
+		LL_PWR_EnableBkUpAccess();
 
-__unused
-static void set_up_csi(void)
-{
-#if defined(STM32_SYSCLK_SRC_CSI) || defined(STM32_PLL_SRC_CSI)
-	/* Enable CSI oscillator */
-	LL_RCC_CSI_Enable();
-	while (LL_RCC_CSI_IsReady() != 1) {
+		/* Enable LSE oscillator */
+		LL_RCC_LSE_Enable();
+		while (LL_RCC_LSE_IsReady() != 1) {
+		}
 	}
-#endif /* STM32_*_SRC_CSI */
 }
 
 __unused
-static int config_src_sysclk_pll(void)
+static int set_up_plls(void)
 {
-#if defined(STM32_SYSCLK_SRC_PLL)
+#if defined(STM32_PLL_ENABLED) || defined(STM32_PLL3_ENABLED)
 	int r;
 	uint32_t vco_input_range;
 	uint32_t vco_output_range;
 
-	/* Configure PLL source */
-	/* Can be HSE , HSI 64Mhz/HSIDIV, CSI 4MHz*/
-	if (IS_ENABLED(STM32_PLL_SRC_HSE)) {
-		set_up_hse();
-
-		/* Main PLL configuration and activation */
-		LL_RCC_PLL_SetSource(LL_RCC_PLLSOURCE_HSE);
-	} else if (IS_ENABLED(STM32_PLL_SRC_CSI)) {
-		set_up_csi();
-
-		/* Main PLL configuration and activation */
-		LL_RCC_PLL_SetSource(LL_RCC_PLLSOURCE_CSI);
-	} else if (IS_ENABLED(STM32_PLL_SRC_HSI)) {
-		set_up_hsi();
-
-		/* Main PLL configuration and activation */
-		LL_RCC_PLL_SetSource(LL_RCC_PLLSOURCE_HSI);
-	} else {
-		return -ENOTSUP;
+	if (IS_ENABLED(STM32_PLL_ENABLED) || IS_ENABLED(STM32_PLL3_ENABLED)) {
```

Comment:
`if (IS_ENABLED(STM32_PLL_ENABLED) || IS_ENABLED(STM32_PLL3_ENABLED)) {`
Is redundant with 
`#if defined(STM32_PLL_ENABLED) || defined(STM32_PLL3_ENABLED)`
few lines above

Reply:
Good catch, removed, thanks